### PR TITLE
Simplify fixtures

### PR DESCRIPTION
**Pull request by hackaugusto**
This PR does:

- Removes the extra tool to `deploy_contract` that does the same things as `JSONRPCClient.deploy_solidity_contract` 
- Removes a duplicated fixtures to deploy these smart contracts:
  - Secret registry
  - Token Network Registry

### CODE REVIEW

File: raiden/tests/integration/contracts/fixtures/contracts.py [hunk 1]
```diff
@@ -13,42 +13,54 @@
...
-            constructor_parameters=args,
-        )
-    return f
-
-
-@pytest.fixture
-def secret_registry_contract(deploy_contract):
-    return deploy_contract(CONTRACT_SECRET_REGISTRY)
+@pytest.fixture(name='secret_registry_contract')
+def deploy_secret_register_and_return_jsonrpc_proxy(deploy_client, contract_manager):
```
Comment (palango) [inline raiden/tests/integration/contracts/fixtures/contracts.py]: - Why set the name over the attribute?
- s/deploy_secret_register/deploy_secret_registry

File: raiden/tests/integration/contracts/fixtures/contracts.py [hunk 2]
```diff
@@ -96,8 +114,8 @@ def token_network_proxy(deploy_client, token_network_contract, contract_manager)
     )
 
 
-@pytest.fixture
-def token_contract(deploy_client, contract_manager):
+@pytest.fixture(name='token_contract')
+def deploy_network_and_return_jsonrpc_proxy(deploy_client, contract_manager):
```
Comment (palango) [inline raiden/tests/integration/contracts/fixtures/contracts.py]: s/deploy_network/deploy_token

File: raiden/tests/integration/fixtures/smartcontracts.py [hunk 3]
```diff
@@ -58,8 +59,8 @@ def endpoint_registry_address(deploy_client, contract_manager) -> typing.Address
     return address
 
 
-@pytest.fixture
-def secret_registry_address(deploy_client, contract_manager) -> typing.Address:
+@pytest.fixture(name='secret_registry_address')
+def deploy_secret_registyr_and_return_address(deploy_client, contract_manager) -> typing.Address:
```
Comment (palango) [inline raiden/tests/integration/fixtures/smartcontracts.py]: /s/registyr/registry

File: raiden/tests/integration/contracts/fixtures/contracts.py (hunk 1 above)
Comment (konradkonrad) [reply raiden/tests/integration/contracts/fixtures/contracts.py]: > Why set the name over the attribute?

It allows for more descriptive function names in fixture implementation, while allowing to _use_ the fixture with a telling argument name.

Comment (hackaugusto) [reply raiden/tests/integration/contracts/fixtures/contracts.py]: @palango it was a suggestion from our test review, please refer to the issue: https://github.com/raiden-network/raiden/issues/3346

File: raiden/tests/integration/contracts/fixtures/contracts.py (hunk 2 above)
Comment (hackaugusto) [reply raiden/tests/integration/contracts/fixtures/contracts.py]: Hummm, weird, it's showing the wrong lines here:

image

**Comment by hackaugusto** [comment]
@palango I have made the changes you asked for

File: raiden/tests/integration/contracts/fixtures/contracts.py (hunk 2 above)
Comment (palango) [reply raiden/tests/integration/contracts/fixtures/contracts.py]: I think I commented on the first commit by error.

File: raiden/tests/integration/contracts/fixtures/contracts.py (hunk 1 above)
Comment (palango) [reply raiden/tests/integration/contracts/fixtures/contracts.py]: It's quite nice, and I didn't know that was possible.

**Pull request merged**